docs/content/develop/resource.md

```diff
@@ -0,0 +1,455 @@
+2. Copy the following template into the new file:
+   ```yaml
+   # Copyright 2023 Google Inc.
+   # Licensed under the Apache License, Version 2.0 (the "License");
+   # you may not use this file except in compliance with the License.
+   # You may obtain a copy of the License at
+   #
+   #     http://www.apache.org/licenses/LICENSE-2.0
+   #
+   # Unless required by applicable law or agreed to in writing, software
+   # distributed under the License is distributed on an "AS IS" BASIS,
+   # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   # See the License for the specific language governing permissions and
+   # limitations under the License.
+
+   --- !ruby/object:Api::Resource
+   # API resource name
+   name: 'ResourceName'
+   # Resource description for the provider documentation.
+   description: |
+     RESOURCE_DESCRIPTION
+   references: !ruby/object:Api::Resource::ReferenceLinks
+     guides:
+      # Link to quickstart in the API's Guides section. For example:
+      # 'Create and connect to a database': 'https://cloud.google.com/alloydb/docs/quickstart/create-and-connect'
+       'QUICKSTART_TITLE': 'QUICKSTART_URL'
+     # Link to the REST API reference for the resource. For example,
+     # https://cloud.google.com/alloydb/docs/reference/rest/v1/projects.locations.backups
+     api: 'API_REFERENCE_URL'
+   # Marks the resource as beta-only. Ensure a beta version block is present in
+   # provider.yaml.
+   # min_version: beta
+
+   # Inserts styled markdown into the header of the resource's page in the
+   # provider documentation.
+   # docs:
+   #   warning: WARNING_MARKDOWN
+   #   note: NOTE_MARKDOWN
+
+   # URL for the resource within the API domain. This should match the
+   # resource's create URL (excluding any query parameters).
+   # Terraform field names enclosed in double curly braces will be replaced
+   # with the field values from the resource.
+   base_url: 'projects/{{project}}/locations/{{location}}/resourcenames'
+   # URL for a created resource within the API domain. This should match
+   # the URL for getting a single resource.
+   # Terraform field names enclosed in double curly braces will be replaced
+   # with the field values from the resource.
+   self_link: 'projects/{{project}}/locations/{{location}}/resourcenames/{{resource_id}}'
+   # URL for importing a resource that already exists in GCP. In general
+   # this will be a list containing self_link. If the resource cannot be read
+   # from GCP, comment this out and set exclude_import: true instead.
+   import_format: ['projects/{{project}}/locations/{{location}}/resourcenames/{{resource_id}}']
```

Comment:
This is used to add custom entries, it's a former override. It's totally redundant with self_link as written and is going to make it more likely that folks introduce incorrect deviations.

Reply:
it looks like the implementation if import_format defaults to `base_url + "{{name}}"`, which doesn't work if you're using a field like `{{resource_id}}`

Reply:
Went ahead and switched to `{{name}}` and removed this.